certification/artifacts/map_writer.go

```diff
@@ -0,0 +1,34 @@
+package artifacts
+
+import (
+	"errors"
+	"io"
+)
+
+var ErrFileAlreadyExists = errors.New("file already exists")
+
+// MapWriter implements an ArtifactWriter storing contents in a map.
+type MapWriter struct {
+	files map[string]io.Reader
+}
+
+// NewMapWriter creates an artifact writer in memory using a map.
+func NewMapWriter() (*MapWriter, error) {
```

Comment:
```suggestion
func NewMapWriter() *MapWriter {
```
Error never returned.